maplibre/src/map_state.rs

```diff
@@ -47,6 +51,10 @@ impl ViewState {
     }
 }
 
+/// Stores the state of the map, dispatches tile fetching and caching, tessellation and drawing.
+///
+/// FIXME: MapState may not follow the Single-responsibility principle, as it not only stores
+/// the state of the map but also the rendering, caching, etc.
```

Comment:
I kind of see `MapState` as the central struct which holds all the schedulers, managers, controllers etc. I agree that some methods in here, probably should not be here. They have been moved here by a previous refactoring.

I think we could add controllers which handle the scheduling of fetching tiles for example.